### ISSUE TITLE
[FEATURE] Retry failed downloads

### ISSUE THREAD

**Issue by baccccccc** [Jules-WinnfieldX/CyberDropDownloader#547]
Imagine I'm downloading over 9000 files. Maybe couple dozen of them failed for whatever reason. And oftentimes, those errors are _transient._ (Such as `DDoS-Guard detected` in case of Bunkr or occasional `Bad Gateway` in case of jpg.church, etc.) Which means if I retry a few minutes later, there's a pretty high chance they will succeed.

But what options do I have today?

- Inspect `Errored_Download_URLs.csv` by hand and download those files manually. This works fine in case there are very few of such errors, and I can easily figure _where_ to download them.
- Restart downloading with `--ignore-history` and just re-download the whole thing. Now, hope that those _previously errored_ files get downloaded successfully this time. And if _more_ errors happen again, it would be other files, downloaded previously. (But figuring this out _for sure_ becomes problematic because `Errored_Download_URLs.csv` gets appended, and there's no timestamp for each entry, nor there is a separation between “first attempt” and “second attempt” sessions. So, you kinda have to remember where it stopped last time, and compare the “first” and “second” attempts manually. Ugh.)
- Restart download using _a different working folder,_ so that there's no history, and `Errored_Download_URLs.csv` is kept separate as well. Similar to the previous option but it gives you slightly more control over results. (You can manually compare two outputs as one does not overwrite or concatenate to the other.)

Clearly, none of these options are great. I would prefer another way to just **retry download** on known _files that failed_ previously. And if download succeeds this time, they get _removed_ from `Errored_Download_URLs.csv`. And if the error persists, it remains in `Errored_Download_URLs.csv` so that I can still investigate manually.

**Comment by Jules-WinnfieldX** [comment]
I'm confused, why run with --ignore-history? or swap directories? Just re-run the program with the same links.

**Comment by baccccccc** [comment]
Hmm, will it retry by default? I was under impression it just skips all the links it tried previously, regardless of whether they succeeded or not. But maybe I'm wrong? Let me try a few more times. Sorry for red herring!

**Comment by Jules-WinnfieldX** [comment]
it only skips completed downloads

**Comment by baccccccc** [comment]
Ok, I tried a few times in different scenarios, and it indeed tries to re-download failed.

But what do I do if I need to re-download some files which were previously downloaded successfully? Imagine I deleted some of them by mistake. Or, what actually happened to me, is a bad hard drive. And some files got corrupt, so I deleted them manually.

Is there a way to say “go check which files are actually missing in the folder, and re-donwload those”?

(This might be a different feature request, and I'm happy to open a different issue if you prefer so.)
